Indoor navigation UI

MCSTEVE1000 opened the pull request:
This PR replaces the hardcoded indoor mapping and direction system with a cleaner and more maintainable approach using MappedIn. 

Changes:

-  Replaced the manually maintained floor plans, graph nodes, and BFS pathfinding logic with MappedIn's web-based indoor mapping platform.
- Removed all custom logic for graph structures, node neighbors, multi-floor routing, and SVG polyline drawing.
-  Integrated MappedIn WebView links for:
      - CC Building
      - Hall Building
      - JMSB
      - VE Building
      - VL Building

- Added a simple UI to select a building and view the corresponding MappedIn map.

![image](https://github.com/user-attachments/assets/ff38d77c-aed4-45e1-b82f-42e4834a8b89)

MCSTEVE1000 commented:
The only remaining issue is connecting floors via elevators and stairs. I’ve already added the necessary components and specified the floor ranges they span, but I'm still encountering problems with the implementation. I’ve reached out to the company for clarification and support, so I’m hoping to resolve this in the next sprint.

MCSTEVE1000 commented:
How it looks in app:

![image](https://github.com/user-attachments/assets/9c2d3c5f-5f0a-44ae-b39b-56107b716346)


After Selecting hall building and finding a route:
![image](https://github.com/user-attachments/assets/0b8ed624-5ce0-44ee-8836-d2eb9854b57a)

mcndl13 reviewed (changes requested):
The CI is failing because your `package.json` and `package-lock.json` are out of sync. Make sure to delete your current `package-lock.json` and then `npm install` up and re-generate it.
![image](https://github.com/user-attachments/assets/0bce9a32-685b-4952-93fd-88afa530149d)

MCSTEVE1000 commented on lost-in-faubourg-app/package.json:
```diff
@@ -1,13 +1,12 @@
     "reset-project": "node ./scripts/reset-project.js",
     "android": "expo run:android",
-    "ios": "expo run:ios",
+    "ios": "npx expo run:ios",
```
This was added by mistake

MCSTEVE1000 commented on lost-in-faubourg-app/package.json:
```diff
@@ -1,13 +1,12 @@
   "version": "1.0.0",
   "scripts": {
-    "start": "expo start",
+    "start": "npx expo start",
```
Also by mistake

MCSTEVE1000 commented on lost-in-faubourg-app/components/IndoorDirectionsScreen.tsx:
```diff
@@ -1,1089 +1,103 @@
+    color: "white",
+    fontSize: 18,
+  },
+  modalHeader: {
```
Added padding in commit https://github.com/mcndl13/SOEN-390-Mini-Capstone/pull/145/commits/9eadbd99995bd3c10a6e42b21518ab6f54d3adc0

MCSTEVE1000 commented on lost-in-faubourg-app/components/IndoorDirectionsScreen.tsx:
```diff
@@ -1,1089 +1,103 @@
-    marginRight: 10,
+    fontWeight: "bold",
+    color: "white",
+
```
Added bottom padding to WebView component in commit https://github.com/mcndl13/SOEN-390-Mini-Capstone/pull/145/commits/9eadbd99995bd3c10a6e42b21518ab6f54d3adc0

MCSTEVE1000 commented:
Fixed all issues with modal and webview spacing in commit https://github.com/mcndl13/SOEN-390-Mini-Capstone/pull/145/commits/9eadbd99995bd3c10a6e42b21518ab6f54d3adc0. Also regenerated my package-lock.json file.

mcndl13 commented:
As a notice, the pipeline is failing because of a change made in the `HomeScreen.tsx` when merging the google calendar integration. The unit test has been fixed on this [branch](https://github.com/mcndl13/SOEN-390-Mini-Capstone/tree/unit-test-2) and will soon be merged into main.

mcndl13 reviewed (approved):
This was a fantastic PR and effort on your guy's part. This will smooth out future refactoring and unit testing for indoor directions. Great Job!

YFrancis10 commented:
Great job refactoring the Indoor Direction screen and improving UI and functionality. This will definitely help with testing and maintainability of the code. You can merge!

MCSTEVE1000 commented:
Also, I'd like to mention that a lot of refactoring was done in this PR:

1. Replaced Complex Logic with WebView Integration
- Refactoring Pattern: Replaced Function with Command
- Explanation: We replaced a large block of tightly coupled UI + navigation logic with an encapsulated MappedIn WebView, essentially delegating pathfinding and floor mapping responsibilities to a specialized tool.


2. Extract Function
- Refactoring Pattern: Extract Function
- Explanation: By moving from manual direction generation to a single WebView modal that loads each building, we drastically reduced duplicated code and isolated responsibilities.

3. Remove Dead Code
- Refactoring Pattern: Remove Dead Code
- Explanation: We removed all the manually created building graph data, floor images, and BFS logic which were no longer in use, keeping the codebase cleaner.

4. Replaced Complex Conditional with Simpler Logic
- Refactoring Pattern: Replace Nested Conditional with Guard Clauses
- Explanation: The conditional logic for routing across buildings, floors, elevators, and stairs was completely removed and replaced with a uniform decision, loading the appropriate building in a MappedIn WebView.

5. Encapsulate Collection
- Refactoring Pattern: Encapsulate Collection
- Explanation: We encapsulated the list of buildings and their corresponding URLs in a clean array (const buildings = [...])—making it easier to manage and iterate over.

6. Inline Class / Collapse Hierarchy
- Refactoring Pattern: Collapse Hierarchy
- Explanation: Previously, the direction logic, helper functions, and UI were deeply nested and modularized across types and files. We collapsed all that into one straightforward UI component.
